src/simple_grid_2d.rs

```diff
@@ -0,0 +1,83 @@
+use crate::location::Int2D;
+
+// A crude implementation of a matrix based grid, for easy access of specific positions.
+pub struct SimpleGrid2D<T: Copy + Clone> {
+    pub locs: Vec<Vec<Option<T>>>,
+    pub width: i64,
+    pub height: i64,
+}
+
+impl<T: Copy + Clone> SimpleGrid2D<T> {
+    pub fn new(width: i64, height: i64) -> SimpleGrid2D<T> {
+        SimpleGrid2D {
+            locs: vec![vec![None; height as usize]; width as usize],
+            width,
+            height,
+        }
+    }
+
+    pub fn get_value_at_pos(&self, pos: &Int2D) -> Option<T> {
+        return self.locs[pos.x as usize][pos.y as usize];
+    }
+
+    pub fn set_value_at_pos(&mut self, pos: &Int2D, value: T) {
+        self.locs[pos.x as usize][pos.y as usize] = Some(value);
+    }
+}
+
+impl<T: Copy + Clone + PartialOrd> SimpleGrid2D<T> {
+    pub fn min(&self) -> Option<T> {
+        let mut min: Option<T> = None;
+        for i in self.locs.iter() {
+            for j in i.iter() {
+                if let Some(pos) = j {
+                    if let Some(actual_min) = min {
+                        if *pos < actual_min {
+                            min = Some(*pos);
+                        }
+                    } else {
+                        min = Some(*pos);
+                    }
+                }
+            }
+        }
+        min
+    }
+
+    pub fn max(&self) -> Option<T> {
+        let mut max: Option<T> = None;
+        for i in self.locs.iter() {
+            for j in i.iter() {
+                if let Some(pos) = j {
+                    if let Some(actual_min) = max {
+                        if *pos > actual_min {
+                            max = Some(*pos);
+                        }
+                    } else {
+                        max = Some(*pos);
+                    }
+                }
+            }
+        }
+        max
+    }
+}
+
+#[cfg(test)]
```

Comment:
move test to the right folder?

Reply:
This test in particular is an unit test, the rust book suggests to put unit test modules in the same file of the code being tested, whereas integration tests go in a tests folder.
https://doc.rust-lang.org/book/ch11-03-test-organization.html